scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -281,6 +281,26 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
     (left, right)
   }
 
+  /**
+   * Partition this SCollection into a map from possible key values to an SCollection of
+   * corresponding elements based on the provided function .
+   *
+   * @param partitionKeys The keys for the output partitions
+   * @param f function that assigns an output partition to each element, should be in the range
+   * of `partitionKeys`
+   * @return partitioned SCollections in a `Map`
+   * @group collection
+   */
+  def partitionToMap[U: Coder](partitionKeys: Set[U], f: T => U): Map[U, SCollection[T]] = {
```

Comment:
rename to `partition`?

Reply:
that is what I wanted to do originally but the compiler didn't like it.

Caused errors for example in the partition by predicate method it didn't know if to use my new one or the original method. Can that be avoided without breaking backwards compatibility?

Reply:
yeah, it will require type ascription, it's ok to add it in this version since already introduces some breaking changes. TBH `partition` should have been curried

Reply:
on a second thought, it might be better to just use similar function signature to what you had.

```scala
def partitionByKey[U: Coder](partitionKeys: Set[U])(f: T => U): Map[U, SCollection[T]] 
```